### PR TITLE
token fix

### PR DESCRIPTION
**Pull request by colman-hartinger**
fixes get token call to bind token and user info to success

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ const _HelperCallbacks = {
         api.user = this.user;
 
         if (success !== undefined) {
-          success.bind(api.user)();
+          success.bind({token: api.token, user: api.user})();
```
Comment (bradymadams) [inline src/api.ts:29]: Could this just be `success.bind(this)();`?

**Review by bradymadams — CHANGES_REQUESTED** [review]
Increment the revision number.

File: src/api.ts (hunk 1 above)
Comment (colman-hartinger) [reply src/api.ts:29]: Wouldn't that bind the entire this state to success?

Comment (bradymadams) [reply src/api.ts:29]: It would, which I believe is just an object with a user and a token attribute.

File: src/api.ts [hunk 2]
```diff
@@ -22,11 +22,8 @@ const _HelperCallbacks = {
       if (this.error && error !== undefined) {
         error.bind(this)();
       } else {
-        api.token = this.token;
-        api.user = this.user;
```
Comment (bradymadams) [inline src/api.ts:26]: This was setting token and user attributes on the `API` object. Aren't those needed/used in the API class?

Comment (colman-hartinger) [reply src/api.ts:26]: my bad, made that change too quickly without realizing their purpose in the function

**Review by bradymadams — APPROVED** [review]

**Pull request merged**